src/Guzzle6HttpAdapter.php

```diff
@@ -28,7 +30,12 @@ class Guzzle6HttpAdapter implements HttpClient, HttpAsyncClient
      */
     public function __construct(ClientInterface $client = null)
     {
-        $this->client = $client ?: new Client();
+        if (!$client) {
+            $handlerStack = new HandlerStack();
+            $handlerStack->push(Middleware::prepareBody(), 'prepare_body');
```

Comment:
as per #13, we keep the prepareBody functionality. contrary to redirect, exception and cookie, it does not change behaviour and there is no reason to not want this. also, if we add php-http plugins that handle content-length, this middleware realizes the value is already set and leaves things alone.